haystack/modeling/training/base.py

```diff
@@ -596,3 +602,111 @@ def _all_ranks_have_data(self, has_data: bool, step: Optional[int] = None):
             return False
         else:
             return True
+
+class DistillationTrainer(Trainer):
+    def __init__(
+        self,
+        model,
+        teacher_model,
+        optimizer,
+        data_silo: DataSilo,
+        epochs: int,
+        n_gpu: int,
+        device,
```

Comment:
I have added the type hints. However, for lr_scheduler I needed to import a private class from pytorch. I'm not sure if that's desirable.

Reply:
You're referring to something like `from torch.optim.lr_scheduler import _LRScheduler` ?

Reply:
Yes, I am.

Reply:
If mypy doesn't complain about the annotation with `_LRScheduler`, which I think it doesn't, we can use it. Otherwise, type ignore is the next best solution here in my opinion.
@tholor What do you think?

Reply:
Yes, the import looks okay to me - if that's the type we expect here it is what it is ;) . 
If mypy complains we can drop it or add the ignore comment